### PR TITLE
test vectors 1/n

### PR DESCRIPTION
**Pull request by franziskuskiefer**
This PR enables the encryption test vectors again. It also fixes some issues along the way.

Note that there are a number of bulk changes in here that we might want to pull out (clippy changes will go away when #1237 is merged).

### CODE REVIEW

File: openmls/src/tree/tests_and_kats/kats/kat_encryption.rs [hunk 1]
```diff
@@ -220,15 +245,15 @@ fn build_handshake_messages(
...
-        MlsMessageHeader {
-            group_id: group.group_id().clone(),
-            epoch: group.context().epoch(),
-            sender: sender_index,
-        },
+        // MlsMessageHeader {
+        //     group_id: group.group_id().clone(),
+        //     epoch: group.context().epoch(),
+        //     sender: sender_index,
+        // },
```
Comment (kkohbrok) [inline openmls/src/tree/tests_and_kats/kats/kat_encryption.rs:256]: Snip?

**Review by kkohbrok — CHANGES_REQUESTED** [review]
Just one nit, then we can get this in.

File: openmls/src/tree/tests_and_kats/kats/kat_encryption.rs (hunk 1 above)
Comment (franziskuskiefer) [reply openmls/src/tree/tests_and_kats/kats/kat_encryption.rs:256]: Thanks, missed this one.

**Review by kkohbrok — APPROVED** [review]
Thanks for starting to breathe life back into the test vectors. I'm going to regret saying this, but it's good to have test vectors back in the test suite.

**Pull request merged**